csm/conf/setup.py

```diff
@@ -592,15 +592,37 @@ def _configure_system_auto_restart(self):
         if any(is_auto_restart_required):
             Log.debug("Updating All setup file for Auto Restart on "
                              "Failure")
-            Setup._update_csm_files("#< RESTART_OPTION >",
+            Setup._update_systemd_conf("#< RESTART_OPTION >",
                                       "Restart=on-failure")
             Setup._run_cmd("systemctl daemon-reload")
 
     @staticmethod
-    def _update_csm_files(key, value):
+    def _is_use_systemd() -> bool:
+        """
+        Check if systemd should be used for the current set up.
+
+        :returns: True if systemd should be used.
+        """
+
+        env_type = Conf.get(const.CONSUMER_INDEX, const.ENV_TYPE_KEY, None)
+        return env_type in [const.HW, const.VM]
+
+    @staticmethod
+    def _copy_systemd_configuration():
+        if not Setup._is_use_systemd():
+            Log.warn('SystemD is not used in this environment and will not be set up')
+            return
+        Setup._run_cmd(f"cp {const.CSM_AGENT_SERVICE_SRC_PATH} {const.CSM_AGENT_SERVICE_FILE_PATH}")
+        Setup._run_cmd(f"cp {const.CSM_WEB_SERVICE_SRC_PATH} {const.CSM_WEB_SERVICE_FILE_PATH}")
```

Comment:
Copying web service file not required. As it will be part of csm-web setup
@soniyamoholkar  Can you comment on this?

Reply:
CSM web does copying at RPM installation phase, like CSM agent has done before this PR. I removed it as well (https://github.com/Seagate/cortx-management-portal/pull/501) and made the copying here for things to work like they used to. Of course the copying would be better in some kind of csm_web setup, but I don't see it in cortx-management-portal.

Reply:
@udayan-y21  Let it be there as it is right now. "Copying web service file not required. As it will be part of csm-web setup". will be taken care as different PRs. 
@avrnv csm_web_setup is completed however the PR is not merged yet (merged in its feature branch) that's why it is not seen in the main.